<div style="background:linear-gradient(135deg,#1e1b4b 0%,#4338ca 55%,#6366f1 100%);border-radius:18px;padding:34px 30px;color:#fff;font-family:Inter,Segoe UI,sans-serif">
  <div style="font-size:12px;letter-spacing:3px;color:#c7d2fe;font-weight:700;text-transform:uppercase">Chapter 35 · Solutions</div>
  <div style="font-size:34px;font-weight:900;line-height:1.1;margin:10px 0 6px">Random Variables, Worked Answers ✅</div>
  <div style="font-size:15px;color:#e0e7ff;max-width:720px;line-height:1.6">Full solutions to the five challenges. Try them yourself first, then compare.</div>
  <div style="margin-top:16px;font-size:13px;color:#cbd5e1">Statistics, Data Science and AI: A Visual Handbook · John Fisher · 2026</div>
</div>

## ⚙️ Setup

In [1]:
import numpy as np
rng = np.random.default_rng(351)
print("ready")

ready


<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#4f46e5;letter-spacing:1px">CHALLENGE 1 · BUILD A PMF</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Number of heads in 3 flips</div>
<div style="color:#4a5578;margin-top:6px">Let X = the number of heads in 3 fair coin flips. Write the PMF and confirm it sums to 1.</div>
</div>

In [2]:
from math import comb
pmf = {k: comb(3,k)*0.5**3 for k in range(4)}
for k,p in pmf.items():
    print(f"P(X={k}) = C(3,{k})/8 = {p:.3f}")
print(f"sum = {sum(pmf.values()):.3f}")

P(X=0) = C(3,0)/8 = 0.125
P(X=1) = C(3,1)/8 = 0.375
P(X=2) = C(3,2)/8 = 0.375
P(X=3) = C(3,3)/8 = 0.125
sum = 1.000


**Answer:** X follows a binomial pattern with PMF P(X=0)=1/8, P(X=1)=3/8, P(X=2)=3/8, P(X=3)=1/8. The counts 1,3,3,1 are row 3 of Pascal's triangle, and they sum to 8/8 = 1.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#2563eb;letter-spacing:1px">CHALLENGE 2 · EXPECTED VALUE</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Average number of heads</div>
<div style="color:#4a5578;margin-top:6px">Using that PMF, compute E[X], then verify by simulating 100,000 sets of 3 flips.</div>
</div>

In [3]:
EX = sum(k*p for k,p in pmf.items())
print(f"E[X] = sum k*P(k) = {EX:.3f}")
sim = rng.integers(0,2,size=(100_000,3)).sum(axis=1)
print(f"simulated mean = {sim.mean():.3f}  (theory 1.5)")

E[X] = sum k*P(k) = 1.500
simulated mean = 1.503  (theory 1.5)


**Answer:** E[X] = 0(1/8) + 1(3/8) + 2(3/8) + 3(1/8) = 12/8 = **1.5**. With a fair coin the expected number of heads in n flips is just n/2, and the simulation confirms it.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#059669;letter-spacing:1px">CHALLENGE 3 · VARIANCE</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Spread of the head count</div>
<div style="color:#4a5578;margin-top:6px">Find Var(X) and SD(X) for the number of heads in 3 flips, using E[X^2] - (E[X])^2.</div>
</div>

In [4]:
EX  = 1.5
EX2 = sum(k**2*p for k,p in pmf.items())
var = EX2 - EX**2
print(f"E[X^2] = {EX2:.3f}")
print(f"Var(X) = {EX2:.3f} - {EX**2:.3f} = {var:.3f}")
print(f"SD(X)  = {var**0.5:.3f}")

E[X^2] = 3.000
Var(X) = 3.000 - 2.250 = 0.750
SD(X)  = 0.866


**Answer:** E[X&sup2;] = 0 + 1(3/8) + 4(3/8) + 9(1/8) = 24/8 = 3. Var(X) = 3 &minus; 1.5&sup2; = **0.75**, and SD &approx; 0.866. (This matches the binomial formula npq = 3 &times; 0.5 &times; 0.5 = 0.75.)

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#db2777;letter-spacing:1px">CHALLENGE 4 · LINEAR TRANSFORM</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">E[aX + b] and Var(aX + b)</div>
<div style="color:#4a5578;margin-top:6px">A game pays 2X + 1 dollars, where X is a single fair die. Find the expected payoff and its variance.</div>
</div>

In [5]:
EX  = 3.5                       # mean of one die
VarX = sum((k-EX)**2/6 for k in range(1,7))
E_pay  = 2*EX + 1
Var_pay = 2**2 * VarX           # b shifts the mean but not the variance
print(f"Var(X) for one die = {VarX:.4f}")
print(f"E[2X+1]   = 2(3.5)+1 = {E_pay:.1f}")
print(f"Var(2X+1) = 4*Var(X) = {Var_pay:.4f}")

Var(X) for one die = 2.9167
E[2X+1]   = 2(3.5)+1 = 8.0
Var(2X+1) = 4*Var(X) = 11.6667


**Answer:** E[2X+1] = 2&middot;E[X] + 1 = 2(3.5) + 1 = **8** dollars. Variance ignores the shift and squares the scale: Var(2X+1) = 2&sup2;&middot;Var(X) = 4 &times; 2.9167 &approx; **11.67**. Adding a constant moves the center; multiplying stretches the spread.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#7c3aed;letter-spacing:1px">CHALLENGE 5 · EXPECTED VALUE DECISION</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Is the bet worth it?</div>
<div style="color:#4a5578;margin-top:6px">A lottery ticket costs $2. It pays $100 with probability 0.01, otherwise nothing. Find the expected net gain. Should you play?</div>
</div>

In [6]:
cost = 2
E_winnings = 100*0.01 + 0*0.99
E_net = E_winnings - cost
print(f"E[winnings] = 100(0.01) = {E_winnings:.2f}")
print(f"E[net gain] = {E_winnings:.2f} - {cost} = {E_net:.2f}")
print("Negative expected value: not worth it as a money decision." if E_net<0 else "Positive EV.")

E[winnings] = 100(0.01) = 1.00
E[net gain] = 1.00 - 2 = -1.00
Negative expected value: not worth it as a money decision.


**Answer:** E[winnings] = 100(0.01) = $1.00, so E[net] = 1.00 &minus; 2.00 = **&minus;$1.00** per ticket. The expected value is negative, so over many tickets you lose about a dollar each on average. Expected value is the standard yardstick for comparing risky choices, the same idea a model uses when it minimizes expected loss.

---
<div style="background:#ffffff;border:1px solid #e6e9f2;border-radius:16px;padding:22px 26px;font-family:Inter,sans-serif">
<div style="font-size:19px;font-weight:800;color:#1a2138">🎉 Nicely done</div>
<div style="color:#4a5578;line-height:1.8;margin-top:8px">You built a PMF, computed an expected value and verified it by simulation, measured variance with the shortcut formula, applied the linear-transform rules, and used expected value to judge a bet. Center and spread, E[X] and Var(X), are the summary of any random variable.</div>
</div>

<div style="text-align:center;color:#8b94b3;font-size:12px;margin-top:16px">Statistics, Data Science and AI: A Visual Handbook · © 2026 John Fisher</div>